addons/website/static/src/js/editor/snippets.editor.js

```diff
@@ -233,6 +319,56 @@ weSnippetEditor.SnippetsMenu.include({
             throw e;
         }
     },
+    /**
+     * @private
+     */
+    _onAnimateTextClick(ev) {
+        const sel = window.getSelection();
+        const editable = this.options.wysiwyg.$editable[0];
+        if (!sel.rangeCount) {
+            return;
+        }
+        const range = getDeepRange(editable, { splitText: true, select: true, correctTripleClick: true });
+        if (this.$currentAnimatedText.length) {
+            this.$currentAnimatedText.contents().unwrap();
+            this.options.wysiwyg.odooEditor.historyResetLatestComputedSelection();
+            this._toggleHighlightAnimatedTextButton();
+            ev.target.classList.remove('active');
+            this.options.wysiwyg.odooEditor.historyStep();
+        } else {
+            if (sel.getRangeAt(0).collapsed) {
+                return;
+            }
+            const animatedTextEl = document.createElement('span');
+            animatedTextEl.classList = 'o_animated_text o_animate o_animate_preview o_anim_fade_in';
+            let $snippet = null;
+            try {
+                range.surroundContents(animatedTextEl);
+                $snippet = $(animatedTextEl);
+            } catch (e) {
+                if (range.commonAncestorContainer.textContent === range.toString()) {
+                    $snippet = $(range.commonAncestorContainer).wrap(animatedTextEl);
+                } else {
+                    $snippet = null;
+                }
```

Comment:
Is this needed? Not sure why the try catch is needed (comments?) but if I reach this code, is not `$snippet` already `null`?

Reply:
It's needed because I have sometimes this error (e.g. when I select a H1 with a span inside it) :
`Failed to execute 'surroundContents' on 'Range': The Range has partially selected a non-Text node.`